File: bwu/binary_tree/104-maximum-depth-of-binary-tree.py
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root is None:
            return 0
            
        return max(self.maxDepth(root.left), self.maxDepth(root.right)) + 1

File: captainhcg/go/104-maximum-depth-of-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    left, right := maxDepth(root.Left), maxDepth(root.Right)
    if right >= left {
        return right + 1
    }
    return left + 1
}
